src/Propel/Runtime/Adapter/Pdo/PdoAdapter.php

```diff
@@ -357,7 +367,7 @@ public function getGroupBy(Criteria $criteria)
     {
         $groupBy = $criteria->getGroupByColumns();
         if ($groupBy) {
-            return ' GROUP BY ' . implode(',', $groupBy);
+            return 'GROUP BY ' . implode(',', $groupBy);
```

Comment:
is it covered on other place? Looks like this space was added in purpose

Reply:
Yes, that formatting is done in SelectQueryBuilder line 63-78;
```php
        $groupBy = $this->adapter->getGroupBy($this->criteria);
        if ($groupBy) {
            $this->criteria->replaceNames($groupBy);
            $sqlClauses[] = $groupBy;
        }

        $havingSql = $this->buildHavingClause($params);
        if ($havingSql) {
            $sqlClauses[] = $havingSql;
        }

        if ($orderBySql) {
            $sqlClauses[] = $orderBySql;
        }

        $sql = implode(' ', $sqlClauses);
```
I really disliked how those spaces were scattered around the codebase